packages/compass-preferences-model/src/renderer-ipc.ts

```diff
@@ -0,0 +1,52 @@
+import ipc from 'hadron-ipc';
+import type {
+  GlobalPreferences,
+  UserPreferences,
+  OnPreferenceValueChangedCallback,
+} from './preferences';
+
+/**
+ * API to communicate with preferences from the electron renderer process.
+ */
+export const preferencesIpc = {
+  savePreferences(attributes: Partial<GlobalPreferences>) {
+    if (typeof ipc?.ipcRenderer?.invoke === 'function') {
+      return ipc.ipcRenderer.invoke('compass:save-preferences', attributes);
+    }
+    return {} as UserPreferences;
+  },
+  getPreferences() {
+    if (typeof ipc?.ipcRenderer?.invoke === 'function') {
+      return ipc.ipcRenderer.invoke('compass:get-preferences');
+    }
+    return {} as GlobalPreferences;
+  },
+  getConfigurableUserPreferences() {
+    if (typeof ipc?.ipcRenderer?.invoke === 'function') {
+      return ipc.ipcRenderer.invoke(
+        'compass:get-configurable-user-preferences'
+      );
+    }
+    return {} as UserPreferences;
+  },
+  onPreferenceValueChanged(
+    preferenceName: string,
+    callback: OnPreferenceValueChangedCallback
+  ) {
+    const listener = (_: Event, preferences: GlobalPreferences) => {
+      if (Object.keys(preferences).includes(preferenceName)) {
+        // TODO: find a proper type.
+        return callback((preferences as any)[preferenceName]);
```

Comment:
This should “just work” with the suggestion above this one:

```suggestion
        return callback(preferences[preferenceName]);
```